Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java

```diff
@@ -164,7 +205,8 @@ public void onClick(View view) {
                 AlertDialog.Builder builder = new AlertDialog.Builder(ReservationActivity.this);
                 //TODO (woongjin) need to refactor this block
                 //now its spaghetti code
-                if(selectedTime != null && ticketCount != 0){
+                if(selectedTime != null && ticketCount != 0

```

Comment:
For the readability, using De Morgan's laws pop out the NOT operation outside
if(!(selectedTime == null || ticketCount == 0 || timeslot.get(selectedTimeSlotIdx).getIsBooked()))

Reply:
I would actually just quit this function entirely if (selectedTime == null || ticketCount == 0 || ...)

Then continue with the graying out operation below - so you do the error handling before, and run the normal logic in the main body.

Reply:
OK I will handle this problem